src/compiler/checker.ts

```diff
@@ -3188,12 +3271,12 @@ namespace ts {
             }
 
             function isAccessible(symbolFromSymbolTable: Symbol, resolvedAliasSymbol?: Symbol, ignoreQualification?: boolean) {
-                return symbol === (resolvedAliasSymbol || symbolFromSymbolTable) &&
+                return (symbol === (resolvedAliasSymbol || symbolFromSymbolTable) || getMergedSymbol(symbol) === getMergedSymbol(resolvedAliasSymbol || symbolFromSymbolTable)) &&

```

Comment:
same question as above -- why not always call *m* ?

Reply:
Mostly just to avoid the calls if they're not needed, under the assumption that the equality check is cheaper than the merged symbol lookup check.